### ISSUE TITLE
Reenable the support for label encoder.

### ISSUE THREAD

**Issue by trivialfis** [dmlc/xgboost#11256]
We used to have sklearn label encoder in the sklearn interface but removed it for consistency and wider data type support. This is a tracking issue to bring it back, but for all interfaces.
